x-pack/metricbeat/module/gcp/metrics/metrics_requester.go

```diff
@@ -131,15 +193,25 @@ func (r *metricsRequester) getFilterForMetric(serviceName, m string) string {
 			// }
 			zone = strings.TrimSuffix(zone, "*")
 			f = fmt.Sprintf("%s AND resource.label.location=starts_with(\"%s\")", f, zone)
+			break
+		}
+
+		if len(r.config.Regions) != 0 {
+			regionsFilter := r.buildRegionsFilter(r.config.Regions, gcp.GKEResourceLabelLocation)
+			f = fmt.Sprintf("%s AND %s", f, regionsFilter)
 		}
 	case gcp.ServicePubsub, gcp.ServiceLoadBalancing, gcp.ServiceCloudFunctions, gcp.ServiceFirestore, gcp.ServiceDataproc:
 		return f
 	case gcp.ServiceStorage:
-		if r.config.Region == "" {
-			return f
+		switch {
+		case r.config.Region != "":
+			f = fmt.Sprintf(`%s AND resource.labels.location = "%s"`, f, r.config.Region)
+		case len(r.config.Regions) != 0:
+			regionsFilter := r.buildRegionsFilter(r.config.Regions, gcp.StorageResourceLabelLocation)
+			f = fmt.Sprintf("%s AND %s", f, regionsFilter)
 		}
 
-		f = fmt.Sprintf(`%s AND resource.labels.location = "%s"`, f, r.config.Region)
+		return f
```

Comment:
Is this return needed? If not I would remove it there is just one return at the end (where we also log the full filter).